### ISSUE TITLE
client connects, but no sharing

### ISSUE THREAD

**Issue by ghost** [input-leap/input-leap#575]
> This issue has been migrated from old Barrier Github repository https://github.com/debauchee/barrier/issues/575
> 
> Issue created on: 2020-02-27 by @timeweaver
> Issue last updated on: 2021-11-05

### Operating Systems ###

Server: RHEL 8

Client: Windows 10

### Barrier Version ###

2.3.2

### Steps to reproduce bug ###

My client connects, log on the client shows connected to server, log on server shows client connecting.  But mouse won't travel between screens.

### Other info ###

This is a first time setup - I disabled selinux and shut down firewalld on my redhat server in order to allow the client to connect.  Not seeing errors in either log.

---

> Commented on: 2020-02-27 by @timeweaver

Ok - I guess you can close this - It just magically started working - I have no idea why.  I disabled ssl on both and nothing changed, so I re-enabled it.  boom.  working.

---

> Commented on: 2020-02-27 by @timeweaver

And now it's not - not idea - help!!

---

> Commented on: 2020-03-01 by @the-wes

Can you provide some log output? Turn it up to debug1 and see if you get any messages when trying to switch screens? Attach a screenshot of your server config?

---

> Commented on: 2020-03-03 by @timeweaver

Ok, here's a log file with debug turned on.  The txt file is the saved config.  I don't see anything showing up when I move my mouse against the edge of the screen like it is even trying to switch.  I've had two occasions where it started working for about 10 minutes or so, then just stopped.  Sadly, I couldn't get it to work during this log capture. 

[barrier.log](https://github.com/debauchee/barrier/files/4283983/barrier.log)

[barrier.txt](https://github.com/debauchee/barrier/files/4284002/barrier.txt)

---

> Commented on: 2020-03-06 by @nsclong

I have the same issue. I am running Debian 10 (server) and Windows 10 (client), both on Barrier version 2.2.0.

The machines are connecting to each other fine and talking to each other. Debug1 level log attached. However mouse just will not share. It stops at the edges of the server screen.

When I move my mouse to the edge of the screen to attempt to move to the client machine, nothing changes in the log.

[barrier-server.log](https://github.com/debauchee/barrier/files/4297587/barrier-server.log)

If it's relevant, server screen size is 4K, whereas client is standard HD.

---

> Commented on: 2020-03-09 by @timeweaver

Turns out my company won't let me use this software anyway.  Thanks for your efforts, I'm gonna go cry in my milk now.

---

> Commented on: 2020-03-10 by @ik3umt

Same issue here, server win10 + barrier 2.3.2 , client latest debian10 + barrier 2.2.0
Client connects to server, both ok on logs, logs tell entering and leaving screens at relative position, keyboard OK but no control over mouse 

---

> Commented on: 2020-03-12 by @nsclong

OK. I had this fixed briefly. I found that I had an old copy of 'Mouse Without Borders' on my Windows 10 machine. After uninstalling and rebooting, Barrier worked once...Debian server connecting to Windows client. But after rebooting both machines again, I can't get the darn thing to work again :(

---

> Commented on: 2020-04-20 by @geertdobbels

Seems to be the same as #589 where I posted, but in the meantime I read this thread and also captured the log file with debug1.

Server : Linux Fedora 31 (installed via flatpak)
Client: Windows 10
Barrier Version 2.3.2 on both computers.

I had barrier working on the same configuration (though I do not remember the barrier version number) for several months without problems but then due to a technical problem, I had to reinstall Fedora31, and since then I have the following issue:
Barrier server clearly sees the client, as can be seen in the logfile.  After changing the log level to Debug1, I only once could see a couple of messages related to leaving the screen on the bottom (which I even do not remember having done), but after that no more messages to that effect appear, no matter how and on which border I try to leave the server screen.
Server has dual screen, but I tried with single screen: no change.
I also tried with and without SSL, no difference.
[barrier.log](https://github.com/debauchee/barrier/files/4503874/barrier.log)

---

> Commented on: 2020-04-20 by @geertdobbels

After starting barrier from the command shell, I saw a message related to a missing wayland-egl plugin, which made me think that wayland might be the problem (wayland is on by default in gnome 3.4 or at least in fedora31).  This made me think that on my previous fedora31 install, I used to login under Xorg gnome, and not under wayland .....
So I logged out and logged back in but under xOrg gnome instead of the default wayland, and now barrier seems to be working.  The question now is how to make barrier working under wayland....

---

> Commented on: 2020-05-01 by @scottajohnson

I had the same problem with a Win7 Server and raspian clients. Scroll lock was turned on and that locked the mouse to the windows screens.

---

> Commented on: 2020-05-07 by @deisi

I can confirm:
Server: 20.04 Ubuntu Xorg
Client: Arch linux Gnome 3.36.1 Wayland

Connection is handled correctly, and scrolling on client in open window on client works. However the mouse pointer on the client is not moved.

```
[2020-05-07T10:58:35] INFO: switch from "serv" to "mtop" at 1919,787
[2020-05-07T10:58:35] INFO: leaving screen
[2020-05-07T10:58:35] INFO: screen "serv" updated clipboard 1
[2020-05-07T10:58:40] INFO: switch from "mtop" to "serv" at 69,1328
[2020-05-07T10:58:40] INFO: entering screen
```

---

> Commented on: 2020-05-12 by @andybiss

I can confirm as well:
Server: 20.04 Ubuntu Desktop (Xorg)
Client: Fedora Workstation 32 (Xorg, installed via flatpak)

connects and enters screen but no mouse or keyboard

---

> Commented on: 2020-05-13 by @marzukia

In the same boat, both machines Fedora32 running x11. Connects and logs reflect correct screen detection but both keyboard and mouse don't work

Edit: seems that during the process of updating my laptop to fedora 32 it reverted back to wayland. Works primo now thanks @simons-public 

---

> Commented on: 2020-05-13 by @simons-public

For any of the people using Wayland in this thread (i.e. RHEL 8 uses Wayland by default now) [Wayland isn't supported by Barrier](https://github.com/debauchee/barrier/projects/1) as of right now.

---

> Commented on: 2020-09-10 by @CurtisDyer

So when I had this issue, it was actually user error. I had pressed scroll lock at some point, which locked the cursor to my server's screen. Just a heads-up for anyone else who's spacing out over the obvious (or maybe it's just me).

---

> Commented on: 2020-09-22 by @p12tic

Closing as a duplicate of #247. The bug for actual Wayland support is in #109.

---

> Commented on: 2020-09-29 by @frohro

This has been plaguing me for a good while.  I have a touch screen, and found that touching the touch screen allowed me to get into the other screen.  Here is the debug messages I was getting:
[2020-09-29T15:43:36] DEBUG1: try to leave "frohro-260" on right
[2020-09-29T15:43:36] DEBUG: locked by mouse buttonID: 0

---

> Commented on: 2021-01-15 by @mic-lachmann

I have the same problem between win10 and OSX. When I assign switch to max screen to a hotkey, it works well, and then I can move my mouse back to win10, but from win10 I can't go to the max screnn

---

> Commented on: 2021-02-08 by @Motazzhd

ok it was working fine with me when it was mac the server for 1 win and 1 other mac both as clients , however i wanted to use the win as server other 2 macs clients, everything connects well but the mouse doesn't travel.

---

> Commented on: 2021-03-30 by @rajeshdgsb

I had this same issue with barrier across a windos10 and Ubuntu20.04 .
Thanks to @CurtisDyer , I found that this was easy to fix by just undoing the scroll lock !

---

> Commented on: 2021-04-09 by @Leathersoup

I experience this issue from time to time.  I believe it is linked to running full screen apps on (in my case the barrier server). When the full screen application is shut down barrier does not seem to recover access to the screen edges.

---

> Commented on: 2021-04-23 by @gogoprog

Experiencing the same issue from Linux to OSX.
It was working fine then I had a connection problem. When this problem was solved the client can connect but the cursor cannot enter to the client.

Scroll lock/unlock does not fix the issue.

---

> Commented on: 2021-06-01 by @ilakast

The accepted answer to https://askubuntu.com/questions/961304/how-do-you-switch-from-wayland-back-to-xorg-in-ubuntu-17-10 solved the issue for me. Thanks to @geertdobbels for the pointer

---

> Commented on: 2021-06-29 by @COUPON4Vitamins

I had the same issue. The culprit was "Scroll Lock." I didn't notice I had it on as the LED lights on my keyboard stopped working. Once I pressed on ScrLK, everything worked flawlessly again.

---

> Commented on: 2021-06-29 by @frohro

I have discovered that this happens when something in the first screen is not completed before I try to go to the second screen.  For example, if I open a menu, but don't select any item, but then try to go to the second screen, it will not allow me to do go there.

---

> Commented on: 2021-09-03 by @andreluiztorres

I've found it works fine when we're running a full-screen program. Minimized doesn't really allow you to hover from one screen to another....

---

> Commented on: 2021-09-20 by @msilmser

scroll lock was the answer for me

---

> Commented on: 2021-11-05 by @henriquemod

> 
> 
> I had the same issue. The culprit was "Scroll Lock." I didn't notice I had it on as the LED lights on my keyboard stopped working. Once I pressed on ScrLK, everything worked flawlessly again.

> 
> 
> scroll lock was the answer for me

Same here

---

> Closed on: 2020-09-22 by @p12tic
